be/src/exec/pipeline/scan/connector_scan_operator.cpp

```diff
@@ -191,15 +192,31 @@ Status ConnectorChunkSource::_read_chunk(RuntimeState* state, vectorized::ChunkP
         return Status::EndOfFile("limit reach");
     }
 
-    do {
-        RETURN_IF_ERROR(_data_source->get_next(state, chunk));
-    } while ((*chunk)->num_rows() == 0);
+    while (_status.ok()) {
+        vectorized::ChunkPtr tmp;
+        _status = _data_source->get_next(state, &tmp);
+        if (_status.ok()) {
+            _ck_acc.push(std::move(tmp));
```

Comment:
if you use ChunkAccumulator in here, I think you need to some optimizations for More than half chunks like ChunkAccumulateOperator.  such as chunk num are: 4000, 3500, 4000, 3500